pkg/stanza/operator/input/windows/xml.go

```diff
@@ -148,19 +148,30 @@ func (e *EventXML) parseMessage() (string, map[string]interface{}) {
 	}
 }
 
-// parse event data entries into a map[string]interface
-// where the key is the Name attribute, and value is the element value
-// entries without Name are ignored
+// parse event data into a map[string]interface
 // see: https://learn.microsoft.com/en-us/windows/win32/wes/eventschema-datafieldtype-complextype
-func parseEventData(entries []EventDataEntry) map[string]interface{} {
-	outputMap := make(map[string]interface{}, len(entries))
+func parseEventData(eventData EventData) map[string]interface{} {
+	outputMap := make(map[string]interface{}, 3)
+	if eventData.Name != "" {
+		outputMap["name"] = eventData.Name
+	}
+	if eventData.Binary != "" {
+		outputMap["binary"] = eventData.Binary
+	}
+
+	if len(eventData.Data) == 0 {
+		return outputMap
+	}
 
-	for _, entry := range entries {
-		if entry.Name != "" {
-			outputMap[entry.Name] = entry.Value
+	dataMaps := make([]interface{}, len(eventData.Data))
+	for i, data := range eventData.Data {
+		dataMaps[i] = map[string]interface{}{
+			data.Name: data.Value,
```

Comment:
I'm not sure this addresses #24493. If I understood correctly, the problem was that `Name` is optional, so we can't just assume it exists. This is why I suggested we represent each entry as a map. e.g.

```suggestion
	for i, data := range eventData.Data {
		dataMaps[i] = map[string]interface{}{
			"name": data.Name,
			"value": data.Value,
```

Reply:
@swiatekm-sumo, @sumo-drosiek, was this your understanding as well?

Reply:
Yes, name is optional, I opted to create the map that way to keep it economical: it creates a single map no matter if the name attribute is set or not. Current code just excludes any Data without name, however, because this change isolates each map as an element in a slice there are no concerns about duplicate keys (which is the reason the current code drops Data elements without name).

Reply:
The reasons we've originally used this format were:

* It's easier to consume as a map
* All the other open source agents I've checked did something similar
* According to the documentation, this should exist for well-formed Events

Maybe we could add a setting controlling it? I know you were against this originally @djaglowski, but it's kind of annoying that there isn't a way to change the format downstream. If we actually want it as a map and only the valid events, we'd essentially need to write a dedicated processor for it.

Reply:
I'm fine with this format. My suggestions were aimed only at ensuring that the data is structurally consistent. It sounds like with this design there is not a _structural_ difference between events with or without names.

Reply:
> According to the documentation, this should exist for well-formed Events

This is not true for the oldest event message format. That one is positional, so the message format string is something like (made up exempla) "Disk %1 has %2 files", and the Data elements won't have the attribute name. If the publisher is correct configured the code will be able to generate the friendly message, however, if it is somehow broken there won't be any useful information on the event.

Reply:
Random example from my box of an event from a service "Google Update Service (gupdate)" which doesn't have a publisher manifest:
```xml
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
 <System>
  <Provider Name="gupdate" /> 
  <EventID Qualifiers="0">0</EventID> 
  <Version>0</Version> 
  <Level>4</Level> 
  <Task>0</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x80000000000000</Keywords> 
  <TimeCreated SystemTime="2023-10-25T16:40:16.1906180Z" /> 
  <EventRecordID>385722</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="0" ThreadID="0" /> 
  <Channel>Application</Channel> 
  <Computer>R91395DV.splunkcorp.com</Computer> 
  <Security /> 
 </System>
 <EventData>
  <Data>Service stopped</Data> 
 </EventData>
</Event>
```
So capturing the data makes clear what happened, ignoring it, doesn't provide any useful information. Even the `eventvwr` just shows something useful on the XML view, the "General" tab shows "The operation completed successfully." since it doesn't have a proper way to render the message.

Reply:
@swiatekm-sumo, @sumo-drosiek, does this work for you?

Reply:
Yeah, I'm ok with it. I'm not happy that we keep changing it and that there's no easy way to change the format downstream, but your proposal is definitely an improvement.